### PR TITLE
detekt (1.1.1) new formula

### PR DESCRIPTION
**Pull request by wasanx25**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

It has been stable version. 
Ref: https://github.com/Homebrew/homebrew-core/pull/24225

@benjamincassidy Thanks, I referenced it.

### CODE REVIEW

File: Formula/detekt.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "detekt-cli-1.1.1-all.jar"
+    bin.write_jar_script libexec/"detekt-cli-1.1.1-all.jar", "detekt"
+  end
+
+  test do
+    system bin/"detekt", "--help"
```
Comment (Bo98) [inline Formula/detekt.rb]: Is it possible you could test actual functionality?

As per the [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook):

> We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Comment (wasanx25) [reply Formula/detekt.rb]: I added fixed of commit.
It is the most simplified test, is this allright?

https://github.com/Homebrew/homebrew-core/pull/46981/commits/b65d87a6eadb8004e148d60267058eaa86c94506

Comment (Bo98) [reply Formula/detekt.rb]: That's fine. It doesn't need to be complicated.

**Review by Bo98 — APPROVED** [review]

File: Formula/detekt.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Detekt < Formula
+  desc "Static code analysis for Kotlin"
+  homepage "https://github.com/arturbosch/detekt"
+  url "https://jcenter.bintray.com/io/gitlab/arturbosch/detekt/detekt-cli/1.1.1/detekt-cli-1.1.1-all.jar"
+  sha256 "8b17c25cd0e9c507773eae0afff97e89c4a1c502b1b1b91c8d55a62032326b3a"
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "detekt-cli-1.1.1-all.jar"
```
Comment (SMillerDev) [inline Formula/detekt.rb]: Please use `#{version}` here instead of the 1.1.1 number so it doesn't need to be changed on every version bump

**Comment by SMillerDev** [comment]
Looks good to me. Thanks for your pull request @wasanx25 without people like you it would be impossible to maintain homebrew to the high standards people have come to expect of the project.

**Comment by chenrui333** [comment]
@BrewTestBot test this please

**Comment by chenrui333** [comment]
Thanks @@wasanx25 !! Appreciate your efforts!

**Comment by wasanx25** [comment]
Thanks reviewers for checking!

**Pull request closed without merging**